### DISCUSSION TITLE
Add a new target to export to CLAP AND VST

### DISCUSSION THREAD

**Discussion by RustoMCSpit** [godotengine/godot-proposals#9431] [Engine Core]
### Describe the project you are working on

I am working on a project to make plugins for audio work. Typically, I'd do this with VST however CLAP has came out and is an open source alternative.

Below is info on clap and frameworks:
https://github.com/free-audio/clap
https://cleveraudio.org/
https://nakst.gitlab.io/tutorial/clap-part-1.html
https://github.com/robbert-vdh/nih-plug

### Describe the problem or limitation you are having in your project

I can't export to the .clap format

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Adding clap export functionality

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

An export to .clap option

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This can be a plugin but I doubt it could be a few lines

### Is there a reason why this should be core and not an add-on in the asset library?

It could be an add-on but an export target is a core type of function

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

**Reply by RustoMCSpit** [reply]
> Moving to discussion, as there is no concrete technical implementation provided.

are the below examples good enough for understanding?

**Comment by RustoMCSpit** [comment]
https://github.com/free-audio/clap/issues/398

**Comment by RustoMCSpit** [comment]
> Moving to discussion, as there is no concrete technical implementation provided.

im no coder, i can only float the basic idea. ive added it to the issues for clap as well. if theres interest in partnering up, thatd be awesome

**Comment by RustoMCSpit** [comment]
> Moving to discussion, as there is no concrete technical implementation provided.

https://github.com/brummer10/XUiDesigner?tab=readme-ov-file
this is what i mean but in godot

**Comment by RustoMCSpit** [comment]
https://hise.dev/
or something like this!

**Comment by djtuBIG-MaliceX** [comment]
This would be something of interest for some audio-bound projects developed in Godot already.  CLAP support is growing steadily for hosts, and the ability to create a native UI of sorts straight from Godot with a purpose-built audio API for real-time use and exported to CLAP for any target platform would be HUGE.

But it probably will need to have things like MIDI and Audio buffer hooks well-defined to tie in with the CLAP API as well, and that appears to be supported through third-party extensions in the C# or C/C++ space.

**Comment by Corruptinator** [comment]
Hey @RustoMCSpit, This might sound random and or breaking the silence but...

I'm thinking maybe the title for the discussion should be updated because Steinberg just made VST3 **_MIT Open Source_**:
https://www.steinberg.net/developers/vstsdk/
(you can fully download the vst3 SDK from the link without any licensing process

From Steinberg website:
_The correct copyright notice for VST is:
MIT License
Copyright (c) 2025 Steinberg Media Technologies GmbH. All rights reserved._

_"Export to both CLAP AND VST"_

This is honestly insane and VST going MIT wasn't on my Open-Source bingo card

**Reply by Corruptinator** [reply]
It's also the same License as the Godot Engine... Maybe now there is a potential for Godot to Export as a VST Plugin. The possibilities are endless mostly cause with the freedom to easily create graphical user interfaces in Godot then using coding underneath it alongside the VST3 SDK it would virtually allow anyone to make any VST plugins to their hearts content.

**Reply by RustoMCSpit** [reply]
done
